src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientHandler.java

```diff
@@ -0,0 +1,121 @@
+
+/** This class handles all calls to manage the streaming ingestion client */
+public class StreamingClientHandler {
+  /**
+   * Checks if a given client is valid (not null, open and has a name)
+   *
+   * @param client The client to validate
+   * @return If the client is valid
+   */
+  public static boolean isClientValid(SnowflakeStreamingIngestClient client) {
+    return client != null && !client.isClosed() && client.getName() != null;
+  }
+
+  private static final KCLogger LOGGER = new KCLogger(StreamingClientHandler.class.getName());
+  private static final String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT_";
+
+  private AtomicInteger createdClientId = new AtomicInteger(0);
+
+  /**
+   * Creates a streaming client from the given config
+   *
+   * @param connectorConfig The config to create the client
+   * @return A newly created client
+   */
+  public SnowflakeStreamingIngestClient createClient(Map<String, String> connectorConfig) {
+    LOGGER.info("Initializing Streaming Client...");
+
+    // get streaming properties from config
+    Properties streamingClientProps = new Properties();
+    streamingClientProps.putAll(
+        StreamingUtils.convertConfigForStreamingClient(new HashMap<>(connectorConfig)));
+
+    try {
+      // Override only if bdec version is explicitly set in config, default to the version set
+      // inside Ingest SDK
+      Map<String, Object> parameterOverrides = new HashMap<>();
+      Optional<String> snowpipeStreamingBdecVersion =
+          Optional.ofNullable(connectorConfig.get(SNOWPIPE_STREAMING_FILE_VERSION));
+      snowpipeStreamingBdecVersion.ifPresent(
+          overriddenValue -> {
+            LOGGER.info("Config is overridden for {} ", SNOWPIPE_STREAMING_FILE_VERSION);
+            parameterOverrides.put(BLOB_FORMAT_VERSION, overriddenValue);
+          });
+
+      String clientName = this.getNewClientName(connectorConfig);
+
+      SnowflakeStreamingIngestClient createdClient =
+          SnowflakeStreamingIngestClientFactory.builder(clientName)
+              .setProperties(streamingClientProps)
+              .setParameterOverrides(parameterOverrides)
+              .build();
+
+      LOGGER.info("Successfully initialized Streaming Client. ClientName:{}", clientName);
+
+      return createdClient;
+    } catch (SFException ex) {
+      LOGGER.error("Exception creating streamingIngestClient");
+      throw new ConnectException(ex);
+    }
+  }
+
+  /**
+   * Closes the given client. Swallows any exceptions
+   *
+   * @param client The client to be closed
+   */
+  public void closeClient(SnowflakeStreamingIngestClient client) {
+    // don't do anything if client is already invalid
+    if (!isClientValid(client)) {
+      return;
+    }
+
+    LOGGER.info("Closing Streaming Client:{}", client.getName());
+    try {
+      client.close();
+    } catch (Exception e) {
+      // the client should auto close, so don't throw an exception here
```

Comment:
Is this true? 

The client will auto close only if its within a try with resources [block](https://www.baeldung.com/java-try-with-resources#replacing).

Reply:
Fair - I'm not sure its true, I can remove that message. I was looking into it as well, and we don't ever use a try with resources block, nor do we do anything additional if the close fails, so I'm not sure why it extends autocloseable if we never use the 'autoclosing'. I don't think it is possible for us to use the client in just a try with resources block

Reply:
Yes we dont need to use it with try with resources and is not possible here because the client is long lived. Auto closeable is useful to not leak resources. (For instance filereader/writer)